MoEPluginBase/MoEPluginController.swift

```diff
@@ -0,0 +1,296 @@
+        }
+    }
+    
+    func resetUser(userAttribute: [String: Any]) {
+        MOAnalytics.sharedInstance.resetUser(forAppID: identifier)
+    }
+    
+    func setUserAttribute(userAttribute: [String: Any]) {
+        guard let dataDict = userAttribute[MoEPluginConstants.General.data] as? [String: Any],
+              let type = dataDict[MoEPluginConstants.General.type] as? String,
+              let attributeName = dataDict[MoEPluginConstants.UserAttribute.attributeName] as? String
+        else
+        {
+            return
+        }
+        
+        let attributeValue = dataDict[MoEPluginConstants.UserAttribute.attributeValue]
+        
+        switch type {
+        case MoEPluginConstants.UserAttribute.general:
+            MOAnalytics.sharedInstance.setUserAttribute(attributeValue, withAttributeName: attributeName, forAppID: identifier)
+            
+        case MoEPluginConstants.UserAttribute.timestamp:
+            if let timeStamp = attributeValue as? String {
+                MOAnalytics.sharedInstance.setUserAttributeISODate(timeStamp, withAttributeName:attributeName,forAppID: identifier)
+            }
+            
+        case MoEPluginConstants.UserAttribute.location:
+            if let locationAttributeDict = dataDict[MoEPluginConstants.UserAttribute.locationAttribute] as? [String: Any],
+               let latitude = locationAttributeDict[MoEPluginConstants.UserAttribute.latitude] as? Double,
+               let longitude = locationAttributeDict[MoEPluginConstants.UserAttribute.longitude] as? Double
+            {
+                MOAnalytics.sharedInstance.setLocation(MOGeoLocation(withLatitude: latitude, andLongitude: longitude), withAttributeName: attributeName, forAppID: identifier)
+            }
+            
+        default:
+            break
+        }
+    }
+    
+    func trackEvent(eventAttribute: [String: Any]) {
+        guard let dataDict = eventAttribute[MoEPluginConstants.General.data] as? [String: Any],
+              let eventName = dataDict[MoEPluginConstants.EventTracking.eventName] as? String
+        else
+        {
+            return
+        }
+        
+        var eventAttributeDict = dataDict[MoEPluginConstants.EventTracking.eventAttributes] as? [String: Any]
+        if let isNonInteractive = dataDict[MoEPluginConstants.EventTracking.isNonInteractive] as? Bool {
+            eventAttributeDict?[MoEPluginConstants.EventTracking.isNonInteractive] = isNonInteractive
+        }
+        
+        let properties = MOProperties()
+        properties.updateAttributes(withPluginPayload: eventAttributeDict)
+        MOAnalytics.sharedInstance.trackEvent(eventName, withProperties: properties, forAppID: identifier)
+    }
+    
+    //MARK: InApp
+    
+    func showInApp(inApp: [String: Any]) {
+        MOInApp.sharedInstance().showInAppCampaign(forAppID: identifier)
+    }
+    
+    func getSelfHandledInApp(inApp: [String: Any]) {
```

Comment:
@RakshithaAcharya Let's avoid using `get` or `set` as prefix for any function names. We can use `fetch` or `update` instead. Let me know your thoughts here

Reply:
@deepa-iOS-SDK  the same name `getselfhandledinapp` is used in native , hybrid. Hence ill be keeping the same name. However I have changed in rest of the place where there is no such dependency